src/components/LocalePicker/index.js

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import {withOnyx} from 'react-native-onyx';
+import PropTypes from 'prop-types';
+import styles from '../../styles/styles';
+import Picker from '../Picker';
+import Text from '../Text';
+import compose from '../../libs/compose';
+import {setLocale} from '../../libs/actions/App';
+import withLocalize, {withLocalizePropTypes} from '../withLocalize';
+import ONYXKEYS from '../../ONYXKEYS';
+import CONST from '../../CONST';
+import Permissions from '../../libs/Permissions';
+
+const propTypes = {
+    /** Indicates which locale the user currently has selected */
+    preferredLocale: PropTypes.string,
+
+    /** Indicates size of a picker component and whether to render the label or not */
+    size: PropTypes.oneOf(['normal', 'small']),
+
+    /** Beta features list */
+    betas: PropTypes.arrayOf(PropTypes.string),
+
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    preferredLocale: CONST.DEFAULT_LOCALE,
+    size: 'normal',
+    betas: [],
+};
+
+const LocalePicker = ({
+    preferredLocale, translate, betas, size,
+}) => {
+    if (!Permissions.canUseInternationalization(betas)) {
+        return null;
+    }
+
+    const localesToLanguages = {
+        default: {
+            value: 'en',
+            label: translate('preferencesPage.languages.english'),
```

Comment:
I think we should keep the language name labels in their original Language. So English would always be "English" spanish will always be "Espagnol" and so on. 
Otherwise say someone changes my app to chinese (or I do by accident) I'm going to have a tough time getting it back to english.

Reply:
Yeah, I think I agree. 
So maybe what we want to do is:
- Add a translation for `languageName` that we add to each translation file
- Do not use the translate method from withLocalize here (but we do want to use it for the label in line 55) and instead use the [translate method directly from the translate file ](https://github.com/Expensify/App/blob/5dc947df8a332d72a3dc38241ab5f9e6211c181d/src/libs/translate.js#L29)
- Pass the as 1st param the locale of the item we are showing (ie: 'en' here and 'es' below)

That way, we ensure we have one language name per language and that each language name shows in their own language.

Reply:
I totally agree, I too get into this kinda situation from time to time.

@iwiznia I have implemented the change, but I completely missed the idea behind your first point. Please take a look at the current version of code and let me know if there is something to be added.

Thanks.